### PR TITLE
Handle torchex.copy_ eqally to prims.copy_

### LINKED ISSUE #930: fuse_bound_symbols skip the side effect of Tensor.copy_
## 🐛 Bug

#920 puts `return` after `prim.copy_`, but this skips `Tensor.copy_`. Since #920, `nvFuserExecutor` no longer forcefully drag `return` to the end of the function body, `Tensor.copy_` gets skipped. This happens when `OperatorExecutor("torch")` is involved together with a fusion executor.

#### Code sample

```py
def f(a):
    a += 1

jitted = thunder.jit(
    f, executors=[thunder.get_executor("nvfuser"), thunder.get_executor("torch")]
)
jitted(torch.tensor(0))
```
```
  ...
  File "/opt/pytorch/lightning-thunder/thunder/executors/nvfuserex_impl.py", line 716, in cse
    assert return_bsym.sym.id == prims.PrimIDs.RETURN
AssertionError
```

## Possible fixes

1. Always put `return` at the end of the function. This used to be done only by `nvFuserExecutor` and lead to errors on other fusion executors, so this fix will involve some refactoring.
2. Handle `Tensor.copy_` symbol in the same way as `prim.copy_`. This seems easy and makes sense since they are the only operators with side effect (as far as I am aware).

### PR DESCRIPTION
**Pull request by shino16**
Fixes #930. This is a follow-up of #920, applying the second fix mentioned in #930.

### CODE REVIEW

File: thunder/executors/data_dependent_partition.py [hunk 1]
```diff
@@ -103,7 +103,8 @@ def __init__(self, trace: TraceCtx):
                 for copy_node in copy_nodes:
                     node.parents.add(copy_node)
                     copy_node.children.add(node)
-            elif bsym.sym.id is PrimIDs.COPY_:
+            # torch executor replaces prims.copy_ with torchex.copy_
+            elif bsym.sym.id in (PrimIDs.COPY_, "copy_"):
```
Comment (crcrpar) [inline thunder/executors/data_dependent_partition.py:107]: could you check if `torchex.copy_.id` instead of "copy_" works?

Comment (shino16) [reply thunder/executors/data_dependent_partition.py]: Thanks, I changed it to comparison between syms.

**Comment by IvanYashchuk** [comment]
Hey @shino16, I'm sorry I missed your original issue (https://github.com/Lightning-AI/lightning-thunder/issues/912). Let's take a step back and reevaluate possible solutions. I don't think data_dependent_partition.py should know anything about copies. One possible solution is to introduce fake dependencies to bound symbols (https://github.com/Lightning-AI/lightning-thunder/issues/912#issuecomment-2271733942).
Let's not merge this PR for now.

**Review by IvanYashchuk — APPROVED** [review]
While a long-term solution is being explored let's get this fix in as it's a small change to what's already in the codebase.

**Review by t-vi — APPROVED** [review]
Stamping, thank you @shino16 @IvanYashchuk @crcrpar

**Pull request merged**